templates/internal/appName/rpc/rpc.go.tpl

```diff
@@ -40,16 +40,38 @@ import (
 	// <</Stencil::Block>>
 )
 
+// GRPCDependencies is used to inject dependencies into the GRPCService service
+// activity. Great examples of integrations to be placed into here would be a database
+// connection or perhaps a redis client that the service activity needs to use.
+type GRPCDependencies struct{
+    // <<Stencil::Block(GRPCDependencies)>>
+{{ file.Block "GRPCDependencies" }}
+	  // <</Stencil::Block>>
+
+    {{- $gRPCDependencyInjection := stencil.GetModuleHook "internal/rpc/gRPCDependencyInjection" }}
+    {{- if $gRPCDependencyInjection }}
+    // dependencies injected by modules
+    {{- range $gRPCDependencyInjection }}
+    {{ . }}
+    {{- end }}
+    // end dependencies injected by modules
+    {{- end }}
+}
+
 // GRPCService is the concrete implementation of the serviceActivity interface
 // which defines methods to start and stop a service. In this case the service
 // being implemented is a gRPC server.
 type GRPCService struct {
 	cfg *Config
+	deps *GRPCDependencies
 }
 
 // NewGRPCService creates a new GRPCService instance.
-func NewGRPCService(cfg *Config) *GRPCService {
-	return &GRPCService{cfg}
+func NewGRPCService(cfg *Config, deps *GRPCDependencies) *GRPCService {
```

Comment:
Can we (i.e. is it possible to) leave the existing signature unmodified and introduce a new one to avoid a breaking change? It would a nicer user experience to have a deprecation period instead of, say, running stencil from some other reason and then being forced to refactor stuff (unless the refactor ends up being trivial)

```
func NewGRPCServiceWithDependencies(cfg *Config, deps *GRPCDependencies) *GRPCService {
```


(same comment applies to HTTPService)

Reply:
This breaking change will only affect 2 services,`mint` and `polaroid`, which is why we opted to *not* do what you're proposing (Jesse already proposed it and we talked about it during a design review meeting this morning).